fluency-fluentd/src/main/java/org/komamitsu/fluency/fluentd/ingester/sender/SSLSocketBuilder.java

```diff
@@ -29,19 +30,21 @@
     private final int port;
     private final int connectionTimeoutMilli;
     private final int readTimeoutMilli;
+    private final SocketFactory sslSocketFactory;
 
-    public SSLSocketBuilder(String host, Integer port, int connectionTimeoutMilli, int readTimeoutMilli)
+    public SSLSocketBuilder(String host, Integer port, int connectionTimeoutMilli, int readTimeoutMilli, SSLSocketFactory sslSocketFactory)
```

Comment:
pass in the socketfactory so the user doesn't have to cast while passing it in, also why not make the type in the class SSlSocketFactory to begin with